ui/src/common/SongDatagrid.js

```diff
@@ -136,7 +136,7 @@ const SongDatagridBody = ({
   )
 
   const firstTracks = useMemo(() => {
-    if (!ids) {
+    if (!ids || !data) {
       return new Set()
```

Comment:
Hey @deluan , I'm not completely sure why the error didn't occur when I tested, but you can close this pr. I shall add this change as well in the next commit.